### PR TITLE
Issue 5497: Support the use of tags in the delegated Spring Kotlin generator.

### PR DESCRIPTION
**Pull request by dumitru-petrusca**
Issue https://github.com/OpenAPITools/openapi-generator/issues/5497

Followed the example of the Java Spring implementation.

There are a lot of changed files in the generated PetStore. It appears the classes were slightly different before. Compared to the Java output and it matches the behavior of this PR. 

Added a test in KotlinSpringServerCodegenTest (delegateWithTags)

**PR checklist**

- [ x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [ x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) before.
- [ x] Run the shell script(s) under `./bin/` (or Windows batch scripts under`.\bin\windows`) to update Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit, and these must match the expectations made by your contribution. You only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the code or mustache templates for a language (`{LANG}`) (e.g. php, ruby, python, etc).
- [ x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.3.x`, `5.0.x`. Default: `master`.
- [ x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

@jimschubert (2017/09) ❤️, @dr4ke616 (2018/08) @karismann (2019/03) @Zomzog (2019/04) @andrewemery (2019/10) @4brunu (2019/11)

### CODE REVIEW

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/KotlinSpringServerCodegen.java [hunk 1]
```diff
@@ -440,7 +451,7 @@ public void processOpts() {
 
     @Override
     public void addOperationToGroup(String tag, String resourcePath, Operation operation, CodegenOperation co, Map<String, List<CodegenOperation>> operations) {
-        if (library.equals(SPRING_BOOT) && this.delegatePattern) {
+        if (library.equals(SPRING_BOOT) && !useTags) {
```
Comment (jimschubert) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/KotlinSpringServerCodegen.java:454]: This changes previous behavior for `library.equals(SPRING_BOOT) && this.delegatePattern` and would be a breaking change. Was this intentional? If so, can you speak toward the intention? To me it looks like `this.delegatePattern` was accidentally removed.

Comment (dumitru-petrusca) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/KotlinSpringServerCodegen.java:454]: Thanks for the comment Jim. The replacement of this.delegatePattern with useTags was intentional. I'm currently using the delegate pattern with tags in my Java project and this change was needed to get the Kotlin generator to produce equivalent code to the Java one. In fact if you look at SpringCodegen.java you can see it does something very similar
{code}
if ((library.equals(SPRING_BOOT) || library.equals(SPRING_MVC_LIBRARY)) && !useTags) {
{code}
I understand that this is a breaking change but I'm not sure we can support tags properly while keeping backwards compatibility.

Comment (jimschubert) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/KotlinSpringServerCodegen.java:454]: Fair enough. I'll mark this PR as a breaking change. Thanks for the explanation.

**Comment by jimschubert** [comment]
This contains a potential breaking change in that the condition:

```
library.equals(SPRING_BOOT) && this.delegatePattern
```

Was changed to:

```
library.equals(SPRING_BOOT) && !useTags
```

This was to match the Java spring generator and appears to have no major impact on the delegate pattern with or without `useTags`.

**Pull request merged**